### DISCUSSION TITLE
what these number mean in json

### DISCUSSION THREAD

**Discussion by quantrpeter** [YosysHQ/yosys#3686] [Q&A] [answered]
Hi, what these numbers mean in json? thanks

![image](https://user-images.githubusercontent.com/50471577/221160052-9ea9b842-3703-4fb9-b086-1c11aa39cf54.png)

**Comment by nakengelhardt** [comment, marked as answer]
That's the source code location where this logic cell comes from: in this case, it was inferred from your input code line 8, characters 8-18 of WireExample.v; and during the flow it was techmapped from coarse-grain to fine-grain with the rule found in [techmap.v, line 289, characters 13-25](https://github.com/YosysHQ/yosys/blob/8216b23fb7b0ee1944403943eb066e0689129ba9/techlibs/common/techmap.v#L289).

(Also, that's the output of write_verilog, not write_json, but the src attributes are the same for both backends.)

**Comment by quantrpeter** [comment]
Hi, but the character positions are wried. please see below

<img width="1575" alt="Screenshot_1" src="https://user-images.githubusercontent.com/50471577/221334649-05eed6de-3639-41c9-acd9-6f14f72f0a04.png">

![image](https://user-images.githubusercontent.com/50471577/221334675-1f516f16-3383-4857-a58a-82cc70bc26b5.png)

**Comment by quantrpeter** [comment]
ok , i understand now. The "col position" shown in vscode is fake. If using tab, it shows 4 cols for 1 tab. thanks @nakengelhardt
